### PR TITLE
Fixing node bucket tests

### PR DESCRIPTION
**Pull request by tpai26**
#### Bug/feature Description
1) test_node_bucket_perf
In this test since all jobs were in the same equivalence class the time required for a scheduling cycle was almost same. To show the considerable improvement of node bucket algorithm, I have added the jobs to different equivalence classes so that the time difference between the the scheduling cycle of the two algorithms is more visible.

2) pbs_node_buckets.py
a) Added test to check - Checkpointed jobs use normal code path
b) To speed up log_match() added n=10000

#### Testing logs/output
[node_buckets_perf.txt](https://github.com/PBSPro/pbspro/files/2443845/node_buckets_perf.txt)
[node_not_buckets.txt](https://github.com/PBSPro/pbspro/files/2443847/node_not_buckets.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_node_buckets.py [hunk 1]
```diff
@@ -423,9 +423,41 @@ def test_not_buckets(self):
...
+        self.du.chown(path=self.chk_file, uid=0, gid=0, sudo=True)
+        c = {'$action': 'checkpoint_abort 30 !' + self.chk_file + ' %sid'}
+        self.mom.add_config(c)
+
+        self.scheduler.set_sched_config({'preempt_order': 'C'})
+        attrs = {'Resource_List.select': '1430:ncpus=1:color=orange',
+                 'Resource_List.place': 'scatter:excl'}
+        j_c1 = Job(TEST_USER, attrs)
+        jid_c1 = self.server.submit(j_c1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid_c1)
```
[the remark this answers is not in the document]
Comment (tpai26) [reply test/tests/functional/pbs_node_buckets.py]: Check added to see the first job follows node bucket code

File: test/tests/performance/pbs_node_buckets_perf.py [hunk 2]
```diff
@@ -46,16 +46,17 @@ class TestNodeBucketPerf(TestPerformance):
...
+                            expect=True)
+        self.colors = \
+            ['red', 'orange', 'yellow', 'green', 'blue', 'indigo', 'violet']
         a = {'resources_available.ncpus': 1, 'resources_available.mem': '8gb'}
         # 10010 nodes since it divides into 7 evenly.
         # Each node bucket will have 1430 nodes in it
         self.server.create_vnodes('vnode', a, 10010, self.mom,
                                   sharednode=False,
-                                  attrfunc=self.cust_attr_func)
+                                  attrfunc=self.cust_attr_func, expect=False)
```
[the remark this answers is not in the document]
Comment (tpai26) [reply test/tests/performance/pbs_node_buckets_perf.py]: I used bhroam's way to see the nodes are added - self.server.expect(NODE, {'state=free': (GE, 10010)})

File: test/tests/functional/pbs_node_buckets.py [hunk 3]
```diff
@@ -444,7 +444,7 @@ def test_not_buckets(self):
         j_c1 = Job(TEST_USER, attrs)
         jid_c1 = self.server.submit(j_c1)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid_c1)
-
+        self.scheduler.log_match(jid_c1 + ';Chunk: ' + '1430:ncpus=1:color=orange', n=10000)
```
Comment (bhroam) [inline test/tests/functional/pbs_node_buckets.py]: No need to plus two static strings together.  Just merge them into one string.

**Comment by agrawalravi90** [comment]
Looks good to me, thanks Tejas

File: test/tests/functional/pbs_node_buckets.py (hunk 3 above)
Comment (tpai26) [reply test/tests/functional/pbs_node_buckets.py]: done. Also added latest test run logs

**Comment by tpai26** [comment]
Added the latest test logs

**Comment by bhroam** [comment]
I merged in the PR

**Pull request merged**